docs/networking.md

```diff
@@ -0,0 +1,114 @@
+<!--[metadata]>
++++
+title = "Networking in Compose"
+description = "How Compose sets up networking between containers"
+keywords = ["documentation, docs,  docker, compose, orchestration, containers, networking"]
+[menu.main]
+parent="smn_workw_compose"
+weight=6
++++
+<![end-metadata]-->
+
+
+# Networking in Compose
+
+By default, Compose sets up a single default [network](http://TODO/docker-networking-docs) for your app. Each container for a service joins the default network and is discoverable via DNS under the service's name.
+
+> **Note:** Your app's network is given the same name as the "project name", which is based on the name of the directory it lives in. See the [CLI docs](cli.md#p-project-name-name) for how to override it.
+
+For example, suppose your app is in a directory called `myapp`, and your `docker-compose.yml` looks like this:
+
+    web:
+      build: .
+      ports:
+        - "8000:8000"
+    db:
+      image: postgres
+
+When you run `docker-compose up`, the following happens:
+
+1. A network called `myapp` is created.
+2. A container is created using `web`'s configuration. It joins the network `myapp` under the name `web`.
+3. A container is created using `db`'s configuration. It joins the network `myapp` under the name `db`.
+
+Each container can now look up the hostname `web` or `db` and get back the appropriate container's IP address. For example, `web`'s application code could connect to the URL `postgres://db:5432` and start using the Postgres database.
+
+Because `web` explicitly maps a port, it's also accessible from the outside world via port 8000 on your Docker host's network interface.
```

Comment:
@dave-tucker It looks like it _is_ in the API now.  (It wasn't when we first started writing the Calico driver, but was added some time later, probably pre DockerCon but hard to say.)  I think that means in theory remote drivers could implement the port mapping.  I don't have a feel for how many have or how people may have interpreted any requirement to do so.  There seems to be little documentation around what libnetwork is expecting drivers to do with the parameter (and no comments in the code either) so your guess is as good as mine.  I think the fact you say it doesn't currently work in multi-host and are proposing to silently attach to the bridge also is an indication greater clarity in this area would be helpful.  (I'm slightly hesitant about endorsing the silently connecting to the bridge workaround without understanding more about exactly how this is intended to work.  e.g. How is the container's namespace configured to ensure traffic goes down the right interface etc.)  Perhaps the discussion @shykes refers to is somewhere public and will help clarify for those of us who aren't already familiar with it?